src/loaders/tvg/tvgTvgLoadParser.cpp

```diff
@@ -0,0 +1,676 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <string.h>
+#include "tvgTvgLoadParser.h"
+
+// Enumeration specifying the loading result
+enum class LoaderResult { InvalidType, Success, SizeCorruption, MemoryCorruption, LogicalCorruption };
```

Comment:
I checked the code, this LoadeResult is useless since your code doesn't handle anything for each return values.
it's nothing more than boolean.

Reply:
This enum is used to return one of three states (not applicable, success, error). It can't be handled the same way with bool.

Reply:
This enum is used to return one of three states (invalid = not applicable, success, error). It can't be handled the same way with bool.

Reply:
@mmaciola so where/how do you use & handle them effectively?

Reply:
See eg. this part:
```
auto result = _parseSceneProperty(block, s.get());
if (result == LoaderResult::InvalidType) result = _parsePaintProperty(block, s.get());

if (result > LoaderResult::Success)
```
If result is invalid (that means not applicable), I try another function (_parsePaintProperty). If result is error (> Success), I  finish function. If both are invalid, I treat it as unsupported and skip block.